commitizen/changelog.py

```diff
@@ -213,7 +213,9 @@ def parse_title_type_of_line(value: str) -> str | None:
     return m.groupdict().get("title")
 
 
-def get_metadata(filepath: str, scheme: VersionScheme = Pep440) -> dict:
+def get_metadata(
+    filepath: str, scheme: VersionScheme = Pep440, encoding: str = "utf-8"
```

Comment:
Is the default `"utf-8"` required here? if so, should we add this default elsewhere as well?

Reply:
Since `scheme` is a keyword parameter, a non-keyword parameter cannot come after it in the parameter list. I added `"utf-8"` so as not reorder the parameters and break backwards compatibility.

Instead, I can import `defaults` and use `encoding` defined there:

```python
from commitizen.defaults import encoding
...

def get_metadata(
    filepath: str, scheme: VersionScheme = Pep440, encoding: str = encoding
) -> dict:
    ...
```

Python still [defaults to the 8-bit 1252 character set encoding on Windows](https://docs.python.org/3/using/windows.html#utf-8-mode). Making `"utf-8"` the default encoding is a good choice and shouldn't be problematic (see `Code that may break` in [PEP 528](https://peps.python.org/pep-0528/) and [PEP 529](https://peps.python.org/pep-0529/) for impetus).

Files opened in binary mode [return contents as `bytes` objects without any decoding](https://docs.python.org/3/library/functions.html#open), so `encoding` wasn't added to those uses of `open` (nor for trivial cases like `commands/init.py/is_python_poetry()`.

I'll add `.encoding` everywhere else where it was missed.

Reply:
Resolved in commit `e9d73c5f`.